extension/source/Controllers/Keyring/KeyringController.ts

```diff
@@ -68,5 +60,49 @@ export default class KeyringController
     }
   }
 
-  // TODO: add methods to sign transactions / messages
+  async signTransactions(address: string, tx: Operation) {
+    const signer = await this._getBLSSinger();
+    const privKey = this._getPrivateKeyFor(address);
+
+    return signer.sign(tx, privKey);
+  }
+
+  async _createAccountAndUpdate(privateKey: string): Promise<string> {
+    const address = await this._getContractWalletAddress(privateKey);
+
+    if (address) {
+      this.update({
+        wallets: [
+          ...this.state.wallets,
+          {
+            privateKey,
+            address: address,
+          },
+        ],
+      });
+    }
+
+    return address;
+  }
+
+  private _getPrivateKeyFor(address: string): string {
+    const keyPair = this.state.wallets.find((x) => x.address === address);
+    if (!keyPair) throw new Error('key does not exist');
+    return keyPair.privateKey;
+  }
+
+  private _getContractWalletAddress(privateKey: string) {
+    const provider = new ethers.providers.JsonRpcProvider(PROVIDER_URL);
```

Comment:
Should this provider come from other controllers?
the `SafeEventEmitterProvider` used by other controllers isn't compatible with this

Reply:
It would be better to use one ethers.js provider and pass that down, as this may create additional socket connections/network request overhead. For now though to get it working should be okay.

Also, you use `CHAIN_RPC_URL` instead of adding a new env var. `PROVIDER_URL === CHAIN_RPC_URL`.
```suggestion
    const provider = new ethers.providers.JsonRpcProvider(CHAIN_RPC_URL);
```

Reply:
it should be the provider coming from other controllers.
it is an eip-1193 compatible provider.